scalafmt-tests/src/test/resources/newlines/alwaysBeforeTopLevelStatements.source

```diff
@@ -709,3 +709,119 @@ end Builder
 
 
 val ul = Builder()
+<<< beforeAll zero, no blank
+newlines.topLevelStatementBlankLines = [
+  { blanks = { before = 1, after = 1, beforeAll = 0 } }
+]
+===
+package x.y.z
+object Foo {
+  def apply[V <: VAType](someProperty: String): SomePropertyBuilder[V] =
+    new SomePropertyBuilder(someProperty)
+}
+>>>
+package x.y.z
+
+object Foo {
+  def apply[V <: VAType](someProperty: String): SomePropertyBuilder[V] =
+    new SomePropertyBuilder(someProperty)
+
+}
+<<< beforeAll negative, has blank
+newlines.topLevelStatementBlankLines = [
+  { blanks = { before = 1, after = 1, beforeAll = -1 } }
+]
+===
+package x.y.z
+object Foo {
+
+  def apply[V <: VAType](someProperty: String): SomePropertyBuilder[V] =
+    new SomePropertyBuilder(someProperty)
+}
+>>>
+package x.y.z
+
+object Foo {
+  def apply[V <: VAType](someProperty: String): SomePropertyBuilder[V] =
+    new SomePropertyBuilder(someProperty)
+
+}
+<<< before negative, topLevelBodyIfMinStatements = before
+newlines.topLevelBodyMinStatements = 1
+newlines.topLevelBodyIfMinStatements = [before]
+newlines.topLevelStatementBlankLines = [
+  { blanks = { before = -1, after = 1 } }
```

Comment:
So here, `before = -1` is being ignored due to `newlines.topLevelBodyIfMinStatements = [before]`

Reply:
correct, that's the test, the higher number wins.